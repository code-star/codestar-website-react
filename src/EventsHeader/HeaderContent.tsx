import React, { SFC } from 'react';
import { Typography, Button, Hidden, withStyles } from '@material-ui/core';
import Container from '../Container/Container';
import { translate, TranslationFunction } from 'react-i18next';
import { Link } from 'react-scroll';
import css from './EventsHeader.module.css';
import { purple } from '@material-ui/core/colors';
import compose from 'recompose/compose';

// TODO improve types by replacing "any"
interface IHeaderContentPropsInner {
	t: TranslationFunction;
	classes: Record<string, string>;
	mEvent: any;
	formattedDate: string;
	children?: React.ReactNode;
}

interface IHeaderContentPropsOuter {
	mEvent: any;
	formattedDate: string;
}

const styles = (theme: any) => ({
	button: {
		color: theme.palette.getContrastText(purple[500]),
		backgroundColor: purple[500],
		'&:hover': {
			color: 'white',
			backgroundColor: purple[700],
		},
	},
});

export const HeaderContent: SFC<IHeaderContentPropsInner> = ({
	t,
	classes,
	mEvent,
	formattedDate,
	children,
}) => {
	return (
		<Container fullHeight center className="mt-5 mt-sm-2 mt-md-0">
			<div className="row">
				<div className="col-12">
					<Typography
						align="center"
						variant="display1"
						className={css.nextEventText}
					>
						{t('OUR_NEXT_EVENT')}
					</Typography>
					<Hidden mdUp>
						<Typography
							align="center"
							variant="display2"
							className={css.nextEventTitle}
						>
							{mEvent.name}
						</Typography>
					</Hidden>
					<Hidden smDown>
						<Typography
							align="center"
							variant="display4"
							className={css.nextEventTitle}
						>
							{mEvent.name}
						</Typography>
					</Hidden>
					<Typography
						gutterBottom
						align="center"
						variant="display2"
						className={css.nextEventText}
					>
						{formattedDate}
					</Typography>
					<div style={{ textAlign: 'center' }}>
						<Button
							color="primary"
							variant="raised"
							href={mEvent.link}
							className={`mr-1 ${classes.button}`}
						>
							{t('SIGN_UP')}
						</Button>
						<Link to="event-details" hashSpy smooth>
							<Button variant="contained">{t('MORE_INFO')}</Button>
						</Link>
						<div className="mt-5">{children}</div>
					</div>
				</div>
			</div>
		</Container>
	);
};

export default compose<IHeaderContentPropsInner, IHeaderContentPropsOuter>(
	translate(['events'], { wait: true }),
	withStyles(styles)
)(HeaderContent);
