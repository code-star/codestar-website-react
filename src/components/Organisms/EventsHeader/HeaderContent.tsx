import React, { FC } from 'react'
import { Typography, Button, Hidden, withStyles, createStyles, Theme } from '@material-ui/core'
import Container from '../../../Container/Container'
import { translate, TranslationFunction } from 'react-i18next'
import { Link } from 'react-scroll'
import { purple } from '@material-ui/core/colors'
import compose from 'recompose/compose'
import { IMeetupEvent } from '../../../containers/EventsContainer/EventsContainer.interfaces'

type PropsInner = {
  t: TranslationFunction
  classes: Record<string, string>
}

type PropsOuter = {
  mEvent: IMeetupEvent
  formattedDate: string
  children?: React.ReactNode
}

type Props = PropsInner & PropsOuter

const styles = (theme: Theme) =>
  createStyles({
    button: {
      color: theme.palette.getContrastText(purple[500]),
      backgroundColor: purple[500],
      '&:hover': {
        color: 'white',
        backgroundColor: purple[700],
      },
    },
    nextEventText: {
      color: 'white',
    },
    nextEventTitle: {
      color: 'white',
      fontWeight: 'bold',
    },
  })

export const HeaderContent: FC<Props> = ({ t, classes, mEvent, formattedDate, children }) => {
  return (
    <Container fullHeight center className="mt-5 mt-sm-2 mt-md-0">
      <div className="row">
        <div className="col-12">
          <Typography align="center" variant="h4" className={classes.nextEventText}>
            {t('OUR_NEXT_EVENT')}
          </Typography>
          <Hidden mdUp>
            <Typography align="center" variant="h3" className={classes.nextEventTitle}>
              {mEvent.name}
            </Typography>
          </Hidden>
          <Hidden smDown>
            <Typography align="center" variant="h1" className={classes.nextEventTitle}>
              {mEvent.name}
            </Typography>
          </Hidden>
          <Typography gutterBottom align="center" variant="h3" className={classes.nextEventTitle}>
            {formattedDate}
          </Typography>
          <div style={{ textAlign: 'center' }}>
            <Button color="primary" variant="contained" href={mEvent.link} className={`mr-1 ${classes.button}`}>
              {t('SIGN_UP')}
            </Button>
            <Link to="event-details" hashSpy smooth>
              <Button variant="contained">{t('MORE_INFO')}</Button>
            </Link>
            <div className="mt-5">{children}</div>
          </div>
        </div>
      </div>
    </Container>
  )
}

export default compose<Props, PropsOuter>(
  withStyles(styles),
  translate(['events'], { wait: true })
)(HeaderContent)
