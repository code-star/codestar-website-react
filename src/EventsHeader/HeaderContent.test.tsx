import React from 'react';
import { HeaderContent } from './HeaderContent';
import renderer from 'react-test-renderer';

const t = () => '';

it('renders the content of the Events header', () => {
	const meetupEvent = {
		name: 'Data Oriented Design with Maxim Zaks',
		link: 'https://www.codestar.nl',
	};
	const formattedDate = '26 september 2018';
	const comp = renderer
		.create(
			<HeaderContent
				t={t}
				classes={{}}
				mEvent={meetupEvent}
				formattedDate={formattedDate}
			>
				<button />
			</HeaderContent>
		)
		.toJSON();
	expect(comp).toMatchSnapshot();
});
