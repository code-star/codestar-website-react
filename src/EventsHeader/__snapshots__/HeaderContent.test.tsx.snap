// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the content of the Events header 1`] = `
<div
  className="
			container
			mt-5 mt-sm-2 mt-md-0
			
			WithTheme-WithWidth-Container---fullHeight-2
			
			
			WithTheme-WithWidth-Container---containerCenter-3
		"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <h1
        className="MuiTypography-root-6 MuiTypography-display1-10 MuiTypography-alignCenter-19"
      >
        
      </h1>
      <h1
        className="MuiTypography-root-6 MuiTypography-display4-7 MuiTypography-alignCenter-19"
      >
        Data Oriented Design with Maxim Zaks
      </h1>
      <h1
        className="MuiTypography-root-6 MuiTypography-display2-9 MuiTypography-gutterBottom-23 MuiTypography-alignCenter-19"
      >
        26 september 2018
      </h1>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <a
          className="MuiButtonBase-root-57 MuiButton-root-31 MuiButton-contained-42 MuiButton-containedPrimary-43 MuiButton-raised-45 MuiButton-raisedPrimary-46 mr-1 undefined"
          href="https://www.codestar.nl"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-32"
          >
            
          </span>
          <span
            className="MuiTouchRipple-root-60"
          />
        </a>
        <a
          className={undefined}
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root-57 MuiButton-root-31 MuiButton-contained-42 MuiButton-raised-45"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-32"
            >
              
            </span>
            <span
              className="MuiTouchRipple-root-60"
            />
          </button>
        </a>
        <div
          className="mt-5"
        >
          <button />
        </div>
      </div>
    </div>
  </div>
</div>
`;
